instructure/canvas-ios
# Discussion redesign introduction

What's new:
- Added a website preview screen to the developer menu. You can enter a custom url and add headers fields to load that url in a webview. This is to support future hybrid website integration tests.
- Added a feature flag called hybrid_discussion_details. If this is turned on along with the web's "Discussions/Announcements Redesign" flag then the app will load a course's discussion details in a web view.

Not yet supported:
- Discussions opened from a course or account group.
- Pull to refresh.
- Course announcements.

refs: MBL-15976
affects: Student, Teacher
release note: none

test plan: 
- Check the website preview screen in the Developer Menu > WebSite Preview.
- Enable discussion redesign flag on web and in the app and enter a course discussion's details screen.

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td><img src="https://user-images.githubusercontent.com/72396990/164258596-380a8a0f-a37d-4fd0-b8c2-60fa2181f20c.PNG"></td>
<td><img src="https://user-images.githubusercontent.com/72396990/164258665-e74f771e-99ca-4087-9ec9-60c020987cad.PNG"></td>
</tr>
<tr>
<td></td>
<td><img src="https://user-images.githubusercontent.com/72396990/164258839-fa532f5a-3ae0-42d0-8396-3e7d32d4e74d.PNG"></td>
</tr>
</table>
